compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.cpp

```diff
@@ -382,6 +382,29 @@ WorkgroupMappingAttr::verify(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
+WorkgroupMappingAttr
+WorkgroupMappingAttr::getAttributeFromMappingId(MLIRContext *context,
+                                                int64_t mappingId) {
+  int64_t linearizedDim = std::max<int64_t>(mappingId - 2, 0);
+  WorkgroupId id =
+      symbolizeWorkgroupId(std::min<uint64_t>(mappingId, 2)).value();
+  return WorkgroupMappingAttr::get(context, id, linearizedDim);
+}
+
+bool WorkgroupMappingAttr::operator<(const WorkgroupMappingAttr &rhs) const {
+  return getMappingId() < rhs.getMappingId();
+}
+
+WorkgroupMappingAttr
+WorkgroupMappingAttr::subtractIfGreaterThan(WorkgroupMappingAttr rhs) const {
```

Comment:
I think you added this helper but forgot to use it in your pattern.

Reply:
I think ill drop this method. Ill think about it. Thanks for flagging.